pkg/k8shandler/cluster.go

```diff
@@ -164,33 +154,39 @@ func (elasticsearchRequest *ElasticsearchRequest) CreateOrUpdateElasticsearchClu
 
 					addNodeState(node, nodeStatus)
 					if err := elasticsearchRequest.setNodeStatus(node, nodeStatus, clusterStatus); err != nil {
-						logrus.Errorf("unable to set node status. E: %s", err.Error())
+						logrus.Warnf("unable to set status for node %q: %s", node.name(), err.Error())
 					}
 
-					elasticsearchRequest.updateMinMasters()
-				}
-
-				// ensure we always have shard allocation to All if we aren't doing an upgrade...
-				if ok, err := esClient.SetShardAllocation(api.ShardAllocationAll); !ok {
-					logrus.Warnf("Unable to enable shard allocation: %v", err)
+					// if we created a node ensure that MinMasters is (n / 2 + 1)
+					if elasticsearchRequest.AnyNodeReady() {
+						elasticsearchRequest.updateMinMasters()
+					}
 				}
 
-				// we only want to update our replicas if we aren't in the middle up an upgrade
-				if err := esClient.UpdateReplicaCount(int32(calculateReplicaCount(elasticsearchRequest.cluster))); err != nil {
-					logrus.Error(err)
-				}
-				if aliasNeededMap == nil {
-					aliasNeededMap = make(map[string]bool)
+				// ensure we always have shard allocation to All if we aren't doing an update...
+				if elasticsearchRequest.AnyNodeReady() {
+					if ok, err := esClient.SetShardAllocation(api.ShardAllocationAll); !ok {
+						logrus.Warnf("Unable to enable shard allocation for cluster %q in namespace %q: %v", elasticsearchRequest.cluster.Name, elasticsearchRequest.cluster.Namespace, err)
```

Comment:
Maybe a place to add status update in future.  it might help identify why a cluster is green/yellow/red

Reply:
we currently have a status field to tell us what the current shard allocation is